### ISSUE TITLE
Add common types for public structs used when generating JSON

### ISSUE THREAD

**Issue by mfcastellani** [hermeznetwork/hermez-node#849]
Related to https://github.com/hermeznetwork/hermez-node/pull/842 and https://github.com/hermeznetwork/hermez-node/issues/850

As the comment by @arnaubennassar we need to improve the structures used when generating JSON. 

Once we've public structs that are able to marshal/unmarshal in the api format, they shouldn't use the apitypes types. Example: EthAddr apitypes.HezEthAddr should be EthAddr ethCommon.Address. 

Reasoning: until now we only cared to fulfill the API format, as it was the main duty, but this is very bad for reusing code on the client side, people don't want tor receive a weird string but an actual Ethereum address (in this example, but idx, bjj, .... also aplies). To do so I suggest hiding this intermediary structs inside the marhsal/unmarshal functions [as I did here](https://github.com/hermeznetwork/hermez-node/pull/841/files#diff-6bb7ed027df678ce6822860d0344e836821a8c7e6ab979291dad26de7e48051aR444)

We should try to use the common package as much as possible to avoid having similar structs that represent the same but require extra parsing and similar boring code. Although this is impossible in many cases due to the difference in content from the API objects and the common structs.

**Comment by mfcastellani** [comment]
Related to https://github.com/hermeznetwork/hermez-node/pull/842 and https://github.com/hermeznetwork/hermez-node/issues/850

As the comment by @arnaubennassar we need to improve the structures used when generating JSON. 

Once we've public structs that are able to marshal/unmarshal in the api format, they shouldn't use the apitypes types. Example: EthAddr apitypes.HezEthAddr should be EthAddr ethCommon.Address. 

Reasoning: until now we only cared to fulfill the API format, as it was the main duty, but this is very bad for reusing code on the client side, people don't want tor receive a weird string but an actual Ethereum address (in this example, but idx, bjj, .... also aplies). To do so I suggest hiding this intermediary structs inside the marhsal/unmarshal functions [as I did here](https://github.com/hermeznetwork/hermez-node/pull/841/files#diff-6bb7ed027df678ce6822860d0344e836821a8c7e6ab979291dad26de7e48051aR444)

We should try to use the common package as much as possible to avoid having similar structs that represent the same but require extra parsing and similar boring code. Although this is impossible in many cases due to the difference in content from the API objects and the common structs.
